### PR TITLE
docs(website): update table doc with beta content

### PR DESCRIPTION
**Pull request by richbachman**
- [x] Add new content to table doc page

### CODE REVIEW

File: packages/paste-website/src/pages/components/table/index.mdx [hunk 1]
```diff
@@ -57,47 +67,79 @@ export const pageQuery = graphql`
...
+- Any component can be placed in a Table, but tables should only contain static textual and numeric data, rather
+  than actionable components like buttons, whenever possible. This is because, if the table is reasonably long—say
+  50 rows—with multiple actions per row—say 4 actions—a keyboard-only user would have to press the tab key 200
+  times to get past that table.
+
+<Callout>
+  <CalloutTitle>🦮 Accessibility insight: navigating tables with a screen reader</CalloutTitle>
+  <CalloutText>
+    <Anchor href="https://youtu.be/tlpQ0d2ADNo" showExternal>
+      Watch this video
```
Comment (TheSisb) [inline packages/paste-website/src/pages/components/table/index.mdx:104]: I'm not sure this link text is great for a11y

Comment (katieporter) [reply packages/paste-website/src/pages/components/table/index.mdx:104]: Good catch @TheSisb. How about: "To learn more about why semantic table markup is so important, watch this <a>demo video by Inclusive Components</a> to see what it's like to explore a data table with a screen reader."

**Review by SiTaggart — APPROVED** [review]

File: packages/paste-website/src/pages/components/table/index.mdx (hunk 1 above)
Comment (katieporter) [reply packages/paste-website/src/pages/components/table/index.mdx]: @richbachman the anchor tags I put around the linked text in my comment disappeared haha sorry! I was thinking the linked text should be "this demo video by Inclusive Components".

File: packages/paste-website/src/pages/components/table/index.mdx [hunk 2]
```diff
@@ -287,58 +343,662 @@ Table rows (`Tr`) can be set with either top, middle, or bottom vertical alignme
...
+    <Tr>
+      <Th>User</Th>
+      <Th>Skills</Th>
+      <Th textAlign="right">Tasks</Th>
+    </Tr>
+  </THead>
+  <TBody>
+    <Tr>
+      <Th scope="row">
+        <Stack orientation="horizontal" spacing="space30">
```
Comment (katieporter) [inline packages/paste-website/src/pages/components/table/index.mdx:689]: @richbachman can these be `space40` instead of `space30`?

Comment (katieporter) [reply packages/paste-website/src/pages/components/table/index.mdx:689]: and `sizeIcon70` for the avatar size?

Comment (richbachman) [reply packages/paste-website/src/pages/components/table/index.mdx:689]: Yep. I'll make that change.

File: packages/paste-website/src/pages/components/table/index.mdx [hunk 3]
```diff
@@ -287,58 +343,662 @@ Table rows (`Tr`) can be set with either top, middle, or bottom vertical alignme
...
+      <Th>Profile URL</Th>
+    </Tr>
+  </THead>
+  <TBody>
+    <Tr>
+      <Th scope="row">Adam Brown</Th>
+      <Td>English, French, Sales, Spanish</Td>
+      <Td>
+        <Anchor href="/getting-started/about-paste">
+          <Truncate>https://paste.twilio.design/getting-started/about-paste</Truncate>
```
Comment (katieporter) [inline packages/paste-website/src/pages/components/table/index.mdx:893]: @richbachman do you know why these links aren't underlined?

File: packages/paste-website/src/pages/components/table/index.mdx [hunk 4]
```diff
@@ -287,58 +343,662 @@ Table rows (`Tr`) can be set with either top, middle, or bottom vertical alignme
...
+##### Table
 
-| Property | Default token | Modifiable? |
-| -------- | ------------- | ----------- |
-| color    | color-text    | No          |
-| width    | 100%          | No          |
+| Property      | Default token                                        | Modifiable? |
+| ------------- | ---------------------------------------------------- | ----------- |
+| border-color  | color-border-lighter                                 | No          |
+| border-radius | border-radius-10                                     | No          |
```
Comment (katieporter) [inline packages/paste-website/src/pages/components/table/index.mdx:967]: No more border radius ;)

File: packages/paste-website/src/pages/components/table/index.mdx (hunk 3 above)
Comment (richbachman) [reply packages/paste-website/src/pages/components/table/index.mdx:893]: Just had the order of the components wrong. Good catch. Will be fixed in my next push: https://share.getcloudapp.com/BluQlg1g

File: packages/paste-website/src/pages/components/table/index.mdx (hunk 4 above)
Comment (richbachman) [reply packages/paste-website/src/pages/components/table/index.mdx:967]: Missed that one. Thanks for catching.

File: packages/paste-website/src/pages/components/table/index.mdx [hunk 5]
```diff
@@ -287,58 +343,662 @@ Table rows (`Tr`) can be set with either top, middle, or bottom vertical alignme
...
-| Property             | Default token                           | Modifiable? |
-| -------------------- | --------------------------------------- | ----------- |
-| background-color     | Even rows: color-background-row-striped | No          |
-| border-bottom-width  | border-width-10                         | No          |
-| border-bottom- color | color-border-lighter                    | No          |
+| Property             | Default token                                   | Modifiable? |
+| -------------------- | ----------------------------------------------- | ----------- |
+| background-color     | Striped even rows: color-background-row-striped | No          |
+| border-bottom-width  | border-width-10                                 | No          |
+| border-bottom- color | color-border-lighter                            | No          |
```
Comment (katieporter) [inline packages/paste-website/src/pages/components/table/index.mdx:999]: Super nit picky, but there's an extra space between the dash and `color` here.

File: packages/paste-website/src/pages/components/table/index.mdx [hunk 6]
```diff
@@ -467,31 +1134,47 @@ const TableExample: React.FC = () => {
...
+##### Tr Props
+
+All the valid HTML attributes for `tr` are supported including the following props:
+
+###### `verticalAlign?: "top" | "middle" | "bottom"`
 
 Sets the vertical alignment of the content within the Table row. Defaults to `middle`.
 
-#### Th Props
+##### Th Props
```
Comment (katieporter) [inline packages/paste-website/src/pages/components/table/index.mdx:1161]: do we want to include the prop here to set the Th scope to row?

Comment (richbachman) [reply packages/paste-website/src/pages/components/table/index.mdx:1161]: Thats part of the "All the valid HTML attributes for `th` are supported...", so I don't think it's necessary. We also call that out in the row headers section.

**Review by TheSisb — APPROVED** [review]

**Pull request merged**